File: backend/src/main/java/top/jionjion/agentdesk/entity/ScheduledTask.java
package top.jionjion.agentdesk.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 定时任务实体
 *
 * @author Jion
 */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "scheduled_tasks", schema = "agent_desk")
public class ScheduledTask {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false, length = 128)
    private String name;

    @Column(length = 512)
    private String description;

    @Column(columnDefinition = "text", nullable = false)
    private String prompt;

    @Column(name = "cron_expression", nullable = false, length = 64)
    private String cronExpression;

    @Column(name = "schedule_label", length = 64)
    private String scheduleLabel;

    @Column(name = "skill_id", length = 64)
    private String skillId;

    /**
     * 可选关联项目ID; 置空后任务按无项目任务运行
     */
    @Column(name = "project_id", length = 32)
    private String projectId;

    /**
     * 目标设备ID; 与 projectId 同时保存, 不在触发时猜测在线设备
     */
    @Column(name = "device_id", length = 128)
    private String deviceId;

    @Column(nullable = false)
    private boolean enabled;

    @Column(name = "created_at", nullable = false)
    private long createdAt;

    @Column(name = "updated_at", nullable = false)
    private long updatedAt;
}

File: frontend/src/renderer/src/types/scheduledTask.ts
/** 定时任务 */
export interface ScheduledTask {
    id: number
    name: string
    description: string | null
    prompt: string
    cronExpression: string
    scheduleLabel: string | null
    skillId: string | null
    /** 可选关联项目ID */
    projectId: string | null
    /** 目标设备ID (与 projectId 同时保存) */
    deviceId: string | null
    enabled: boolean
    createdAt: number
    updatedAt: number
}

/** 定时任务创建/更新表单 */
export interface ScheduledTaskFormData {
    name: string
    description?: string
    prompt: string
    cronExpression: string
    scheduleLabel?: string
    skillId?: string
    /** 可选关联项目ID */
    projectId?: string
    /** 目标设备ID, 绑定项目时必填 */
    deviceId?: string
}

/** 定时任务执行记录 */
export interface ScheduledTaskLog {
    id: number
    taskId: number
    taskName: string
    status: 'SUCCESS' | 'FAILED' | 'RUNNING'
    result: string | null
    duration: number
    startedAt: number
    endedAt: number | null
}
